lib/galaxy/datatypes/media.py

```diff
@@ -0,0 +1,214 @@
+            return 'flv' in metadata['format_name'].split(',')
+        except subprocess.CalledProcessException:
+            return False
+
+
+Binary.register_sniffable_binary_format("flv", "flv", Flv)
+
+
+class Mpg(Video):
+    file_ext = "mpg"
+
+    def sniff(self, filename):
+        try:
+            metadata, streams = ffprobe(filename)
+            return 'mpegvideo' in metadata['format_name'].split(',')
+        except subprocess.CalledProcessException:
+            return False
+
+
+Binary.register_sniffable_binary_format("mpg", "mpg", Mpg)
+
+
+class Mp3(Audio):
+    """Class that reads MP3 audio file
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> fname = get_test_fname('audio_2.mp3')
+    >>> Mp3().sniff(fname)
+    True
+    >>> fname = get_test_fname('audio_1.wav')
+    >>> Mp3().sniff(fname)
+    False
+    """
+    file_ext = "mp3"
+
+    def sniff(self, filename):
+        try:
+            metadata, streams = ffprobe(filename)
+            return 'mp3' in metadata['format_name'].split(',')
+        except subprocess.CalledProcessException:
+            return False
+
+Binary.register_sniffable_binary_format('mp3', 'mp3', Mp3)
+
+
+class WAV(Binary):
+    """Class that reads WAV audio file
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> fname = get_test_fname('hello.wav')
+    >>> WAV().sniff(fname)
+    True
+    >>> fname = get_test_fname('audio_1.wav')
+    >>> WAV().sniff(fname)
+    True
+    >>> fname = get_test_fname('audio_2.mp3')
+    >>> WAV().sniff(fname)
+    False
+    """
+
+    file_ext = "wav"
+    blurb = "RIFF WAV Audio file"
+    is_binary = True
+
+    MetadataElement(name="rate", desc="Sample Rate", default=0, no_value=0, readonly=True, visible=True, optional=True)
+    MetadataElement(name="nframes", desc="Number of Samples", default=0, no_value=0, readonly=True, visible=True, optional=True)
+    MetadataElement(name="nchannels", desc="Number of Channels", default=0, no_value=0, readonly=True, visible=True, optional=True)
+    MetadataElement(name="sampwidth", desc="Sample Width", default=0, no_value=0, readonly=True, visible=True, optional=True)
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'audio/wav'
+
+    def sniff(self, filename):
+        """
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('hello.wav')
+        >>> WAV().sniff(fname)
+        True
+
+        >>> fname = get_test_fname('drugbank_drugs.cml')
+        >>> WAV().sniff(fname)
+        False
+        """
+
+        try:
+            fp = wave.open(filename, 'rb')
+            fp.close()
+            return True
+        except wave.Error:
+            return False
+
+    def set_meta(self, dataset, overwrite=True, **kwd):
+        """Set the metadata for this dataset from the file contents
+        """
+
+        fd = wave.open(dataset.dataset.file_name, 'rb')
```

Comment:
If there is a context manager available we should use it, otherwise this should be wrapped in try/finally